### DISCUSSION TITLE
2021-09-07 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#644] [Maintainer discussions]
### Discussion topics

- **[Updates](https://github.com/centerofci/mathesar/discussions/644#discussioncomment-1293403)**: General updates everyone should be aware of.
- **[Questions](https://github.com/centerofci/mathesar/discussions/644#discussioncomment-1293414)**: Questions for the team to answer.
- **[Active Work](https://github.com/centerofci/mathesar/discussions/644#discussioncomment-1293418)**: Work planned for the next week.
- **[Retrospective](https://github.com/centerofci/mathesar/discussions/644#discussioncomment-1293419)**: Summary of last week's work.
- **[Editable Tables & deadline updates](https://github.com/centerofci/mathesar/discussions/644#discussioncomment-1301672)**

### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

**Comment by kgodey** [comment]
## Updates

### Progress
- [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4) is due on Friday.
- [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34) is now mostly fleshed out and we're making progress on backend stability work with two major reorganization pull requests merged this week.
- [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8) has only frontend issues remaining and the first of those was completed this week.
- [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) is also moving along with both frontend and backend issues in progress.
- [09. Working with Views](https://github.com/centerofci/mathesar/milestone/13) is in the design stage and three design specs are in the review process.

### State of Mathesar meetings
The core team had two long "State of Mathesar" meetings in the last week where we reviewed the progress we've made so far and how we've been working together. This has resulted in several changes to the roadmap and processes. You can see the full details in the meeting notes [on our wiki](https://wiki.mathesar.org/meeting-notes/september-2021), but highlights include:

- The alpha release roadmap has been updated to remove high-cost features including visualizations, product-wide search, real time collaboration, data export, improved spreadsheet-style editing, undo & redo for data modeling, and finalization of our API.
- We've added a couple of things to the alpha release plan, including public sharing of views and forms, a better editing experience, and time for user testing and launch preparation.
- We've created a plan and timeline for a beta (public) release.
- We've updated our standup format, GitHub workflow, and design review process.

### Contributors
- We have one new contributor with a merged pull request, @seancolsen!
- @powellc had his first PR merged in, closely followed by a second one.
- @linzhou-zhong and @mr-gabe49 continued to have new PRs merged in.

Thanks for your work. :)

**Comment by kgodey** [comment]
## Questions

@mathemancer, @powellc, @ghislaineguerin, @pavish: Your responses are required. Others are also welcome to respond.

### Retrospective
- What went well over the past week?
- What could have gone better over the past week?
- Are there any improvements you'd like to suggest?
- If you attended the State of Mathesar meetings, was it useful? What did you get out of it?

### Active Work
- Does your workload for this week look okay?
- We have one milestone due in the next week. Are there any blockers to meeting the deadline?

### Discussion
Please respond to the [Updates](https://github.com/centerofci/mathesar/discussions/614#discussioncomment-1259664) post directly if you'd like to discuss anything there. If you have any other topics to discuss, please post them as new replies to the [parent post](https://github.com/centerofci/mathesar/discussions/614#discussion-3548369).

**Reply by ghislaineguerin** [reply]
> What went well over the past week?

I think the shorter deadline for reviews is working well, feedback has been timely and helped me inform the design as I was working on the next issue.

> What could have gone better over the past week?

I think overall it was good. Can't think of any specific improvement, other than to be more mindful of checking links and content of things I'm posting to catch any issues. 

> Are there any improvements you'd like to suggest?

None at the time.

> If you attended the State of Mathesar meetings, was it useful? What did you get out of it?

It was helpful to align as a team. It was great to hear feedback regarding the design process as well.

> Does your workload for this week look okay?

It does

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

None

**Reply by mathemancer** [reply]
> What went well over the past week?

Getting back into the swing of things.  It's good to catch up with the team.

> What could have gone better over the past week?

I was considerably less productive than I'd hoped over the last week.  I also realized partway through my work that there was much more to be done to implement the `MATHESAR_TYPES.MONEY` type than had been needed for the `MATHESAR_TYPES.EMAIL` type.

> Are there any improvements you'd like to suggest?

Nothing that we didn't come up with during the State of Mathesar meetings.

> If you attended the State of Mathesar meetings, was it useful? What did you get out of it?

I think they were useful.  Objectively, we ended up with a list of action items that I don't think we would have arrived at any other way.

> Does your workload for this week look okay?

It's a lot, but I'll start at the top and work my way through things.

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

None that I see.

**Reply by pavish** [reply]
> What went well over the past week?

I started on some refactors I've been wanting to do for a while. I've also been sticking to personal schedules, only focusing on the tasks I've mentioned on the standup, each day.

> What could have gone better over the past week?

I underestimated the effect of refactoring the stores. It touches a lot of areas, and not having tests for the frontend is slowing me down. All the more reason to write tests with maximum coverage, in the stability milestone.

> Are there any improvements you'd like to suggest?

No. I like the dependencies section in our standups, works great for me for reviewing designs.

> If you attended the State of Mathesar meetings, was it useful? What did you get out of it?

I found it very useful. It gave an overall view/clarity on our current state. I am hoping we don't have to cut down more features, but based on the UI progress, I guess we may have to.

> Does your workload for this week look okay?
> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

It will probably take this week to complete the refactoring. In the meanwhile, I do not want to merge other frontend PRs, since this changes some fundamental approaches.

Editing and adding new rows should not take a lot of time. I'm estimating 2 to 4 days for both combined. It would be better if we move the milestone's deadline to next week, 15th.

**Reply by kgodey** [reply]
> What went well over the past week?

I'm glad to be refactoring the `db` code, it will be easier to work with in the future and hopefully more approachable to new contributors. I also think the State of Mathesar meetings went well, it was good to catch up with the team and get on the same page.

> What could have gone better over the past week?

The `db` code reorganization is taking me longer than I thought.

> Are there any improvements you'd like to suggest?

I'm trying out starting a new topic in this discussion to branch off of questions answered here.

> If you attended the State of Mathesar meetings, was it useful? What did you get out of it?

It was useful for me. I think our roadmap is more cohesive and we had a chance to reflect on how we work together and how to improve it.

> Does your workload for this week look okay?

I'm probably not going to be able to finish everything because of the `db` code organization taking longer, but that's okay.

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

Addressed in https://github.com/centerofci/mathesar/discussions/644#discussioncomment-1301672

**Reply by powellc** [reply]
> What went well over the past week?

Not a lot. Having Monday off threw me off to start the week and then a number of things outside of work conspired to make this a really frustrating week for getting things done. Will look to open a PR for more testing around API column alterations tomorrow and get a start on adding numeric ranges. 🤞🏽 

> What could have gone better over the past week?

Lost a lot of focus with the short week due to the US holiday and non-work issues. Hoping to be productive tomorrow.

> Are there any improvements you'd like to suggest?

Not at the moment

> If you attended the State of Mathesar meetings, was it useful? What did you get out of it?

N/A ... ended up having other meetings booked over the top :(

> Does your workload for this week look okay?

Given my lack of focus, no, but I'll do my best to dig out with the last day of the week

> We have one milestone due in the next week. Are there any blockers to meeting the deadline?

I don't see any

**Comment by kgodey** [comment]
## Active Work

This post lists work planned for this week (see also: [Active](https://github.com/centerofci/mathesar/projects/1)). It also includes a current summary of milestones involved in this week's work.

### [2021-09-10] [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[8 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2205.%20Editable%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #195
- **Issues not yet started:**
  - [@pavish] #197

### [2021-09-17] [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **[2 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2206.%202021-09%20Stability%22) already closed.** 
- **Issues currently in progress:**
  - [@kgodey] #582
  - [@andyharley-cci] #225
- **Issues not yet started:**
  - [@kgodey] #592
  - [@kgodey] #280
  - [@kgodey] #623
  - [@kgodey] #583
  - [@ghislaineguerin] #473
  - [@ghislaineguerin] #636
  - [@ghislaineguerin] #638
  - [@pavish] #639
  - [@pavish] #640
  - [@pavish] #641
  - [@pavish] #642
  - [@pavish] #643
  - [@pavish] #213
- **Issues remaining in backlog:**
  - 4 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2206.%202021-09%20Stability%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2206.%202021-09%20Stability%22) issues
  - 2 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2206.%202021-09%20Stability%22) issues

### [2021-09-23] [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[20 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2207.%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@dmos62] #498
  - [@pegnott] #495
- **Issues not yet started:**
  - [@mr-gabe49] #448
- **Issues remaining in backlog:**
  - 11 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2207.%20Working%20with%20Tables%22) issues

### [2021-09-27] [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[27 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2208.%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@powellc] #385
  - [@powellc] #387
  - [@kgodey] #392
- **Issues not yet started:**
  - [@kgodey] #552
  - [@powellc] #384
  - [@mathemancer] #412
  - [@mathemancer] #424
  - [@mathemancer] #430
  - [@mathemancer] #420
  - [@powellc] #410
  - [@mathemancer] #404
  - [@mathemancer] #405
  - [@mathemancer] #407
- **Issues remaining in backlog:**
  - 19 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2208.%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2208.%20Initial%20Data%20Types%22) issues

### [2021-09-30] [2021-09 improvements](https://github.com/centerofci/mathesar/milestone/30)

- **[5 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%222021-09%20improvements%22) already closed.** 
- **Issues currently in progress:**
  - [@asharonbaltazar] #594
  - [@arvind-1] #555
  - [@andyharley-cci] #421
  - [@mr-gabe49] #608
- **Issues remaining in backlog:**
  - 10 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%222021-09%20improvements%22) issues
  - 3 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%222021-09%20improvements%22) issues

### [2021-10-04] [09. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2209.%20Working%20with%20Views%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #456
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243
- **Issues not yet started:**
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #443
- **Issues remaining in backlog:**
  - 7 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2209.%20Working%20with%20Views%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2209.%20Working%20with%20Views%22) issues
  - 1 other issues

### [2021-10-18] [11. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **[7 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2211.%20Multiple%20Database%20Support%22) already closed.** 
- **Issues currently in progress:**
  - [@aaditkamat] #395
- **Issues remaining in backlog:**
  - 3 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2211.%20Multiple%20Database%20Support%22) issues

**Comment by kgodey** [comment]
## Retrospective

This post lists issue updates from 2021-08-31 until now by milestone.

### [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **Issues closed:**
  - [@kgodey] #358

### [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues started, not yet finished:**
  - [@pavish] #195

### [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **Issues closed:**
  - [@pavish] #626
  - [@kgodey] #510
- **Issues started, not yet finished:**
  - [@kgodey] #582
  - [@andyharley-cci] #225
- **New issues created:**
  - #643, created by @kgodey
  - #642, created by @kgodey
  - #641, created by @kgodey
  - #640, created by @kgodey
  - #639, created by @kgodey
  - #638, created by @kgodey
  - #637, created by @kgodey
  - #636, created by @kgodey
  - #635, created by @kgodey
  - #631, created by @seancolsen
  - #626, created by @eito-fis
  - #623, created by @kgodey
  - #619, created by @pavish

### [07. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues closed:**
  - [@seancolsen] #444
- **Issues started, not yet finished:**
  - [@dmos62] #498
  - [@pegnott] #495

### [08. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues closed:**
  - [@powellc] #388
  - [@ghislaineguerin] #241
  - [@ghislaineguerin] #240
- **Issues started, not yet finished:**
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@powellc] #385
  - [@powellc] #387
  - [@kgodey] #392

### [2021-09 improvements](https://github.com/centerofci/mathesar/milestone/30)

- **Issues closed:**
  - [@mr-gabe49] #609
  - [@linzhou-zhong] #517
  - [@mr-gabe49] #474
  - [@linzhou-zhong] #270
- **Issues started, not yet finished:**
  - [@asharonbaltazar] #594
  - [@arvind-1] #555
  - [@andyharley-cci] #421
  - [@mr-gabe49] #608
- **New issues created:**
  - #632, created by @dmos62

### [09. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **Issues started, not yet finished:**
  - [@ghislaineguerin] #456
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243

### [11. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **Issues started, not yet finished:**
  - [@aaditkamat] #395

**Comment by kgodey** [comment]
## Editable Tables & deadline updates

Starting a new topic to discuss @pavish's reply here: https://github.com/centerofci/mathesar/discussions/644#discussioncomment-1297772

> It will probably take this week to complete the refactoring. In the meanwhile, I do not want to merge other frontend PRs, since this changes some fundamental approaches.
>
> Editing and adding new rows should not take a lot of time. I'm estimating 2 to 4 days for both combined. It would be better if we move the milestone's deadline to next week, 15th.

- I've updated the "Editable Tables" milestone deadlines to be the 15th.
- I've also updated the next few milestones so that the deadlines are all clustered towards the end of September and early October. We should be aiming to finish these much earlier, but then I don't have to track/move them on a day-by-day basis. Please see [the roadmap](https://github.com/centerofci/mathesar/milestones?direction=asc&sort=due_date&state=open)

@pavish can you create a new issue to track refactoring work and put it in [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34)? Also, would you be able to get that merged in before editing/adding so that other frontend PRs are unblocked?

**Reply by pavish** [reply]
@kgodey I've created https://github.com/centerofci/mathesar/issues/651, and placed it in  [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34).

I can get this merged in before editing/adding.
